ACG/AutoRouterShield.py

```diff
@@ -0,0 +1,714 @@
+from .AyarLayoutGenerator import AyarLayoutGenerator
+from .Rectangle import Rectangle
+from .XY import XY
+from typing import Tuple, Union, Optional, List, Dict
+from .AutoRouter import EZRouter
+
+
+class EZRouterShield(EZRouter):
```

Comment:
We should try and incorporate these functions directly into the EZRouter class, since most of the methods seem to be pretty similar.